textattack/search_methods/greedy_word_swap_wir.py

```diff
@@ -69,12 +71,47 @@ def _get_index_order(self, initial_text):
                 delta_ps.append(max_score_change)
 
             index_scores = softmax_saliency_scores * np.array(delta_ps)
+
         elif self.wir_method == "delete":
             leave_one_texts = [
                 initial_text.delete_word_at_index(i) for i in range(len_text)
             ]
             leave_one_results, search_over = self.get_goal_results(leave_one_texts)
             index_scores = np.array([result.score for result in leave_one_results])
+
+        elif self.wir_method == "gradient":
+            victim_model = self.get_model()
+            index_scores = np.zeros(initial_text.num_words)
+            grad_output = victim_model.get_grad(initial_text.tokenizer_input)
+            gradient = grad_output["gradient"]
+            j = 0
+            last_matched = 0
+            for i, word in enumerate(initial_text.words):
+                word = initial_text.words[i].lower()
+                matched_tokens = []
+                a = []
+                while j < len(grad_output["tokens"]) and len(word) > 0:
+                    token = grad_output["tokens"][j].lower()
+                    # remove "##" if it's a subword
+                    token = token.replace("##", "")
+                    idx = word.find(token)
+                    if idx == 0:
+                        word = word[idx + len(token) :]
+                        matched_tokens.append(j)
+                        a.append(token)
+                        last_matched = j
+                    j += 1
```

Comment:
This is my biggest comment: I think that this model-specific string processing should happen in the model wrappers. 

Otherwise, we'll find ourselves updating this bit of code every time we add a model with different tokenization specifics (for example, a char-level model, or a model that marked subwords with ** instead of ##). What do you think?

Reply:
I agree.

Reply:
I thought about this more. I think this also works with char-level tokenization since we can treat characters as subwords too. Now regarding different subword prefixes, I'm not sure there's a good way to resolve this since it's something that depends on the model, and doing it in the model wrapper is not gonna be resolve this since model wrapper doesn't know what prefix model uses. 

To address this however, I moved the alignment step to `AttackedText` and moved tokenization + stripping of prefixes to model wrapper.

Reply:
Looks great!